### DISCUSSION TITLE
How do you read and edit a struct variables?

### DISCUSSION THREAD

**Discussion by MrMidnightOfficial** [vlang/v#14901] [General]
How do you read and edit a struct variables?

**Comment by JalonSolov** [comment]
You access them to read them, and assign values to them to "edit".

**Reply by MrMidnightOfficial** [reply]
Example pls

**Reply by Wajinn** [reply]
> Example pls

The documentation (https://github.com/vlang/v/blob/master/doc/docs.md#structs) has good examples.  Maybe you could reference what you read in the documentation that might be confusing for you.

**Comment by Wajinn** [comment]
A basic example would be as below, in terms of defining the struct (outside of the function), and then initialize, assign, and edit inside of a function.  Inside of a function, after the struct fields are assigned values, you can then read or edit them with dot notation.  So to change Mike's name to Tom, would be `student.name = 'Tom'`.  Below, we changed Mike's age from 15 to 16,` student.age = 16`.

```
struct Student {
	mut:
	name string
	age int
	major string
}

fn main() {
	mut student := Student{ name:'Mike', age:15, major:'Math' }

	println(student)
	student.age = 16
	println(student)
	println(student.age)
}
```
